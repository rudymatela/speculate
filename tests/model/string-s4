_ :: [Char]  (holes: [Char])
_ :: [[Char]]  (holes: [[Char]])
"" :: [Char]
" " :: [Char]
"\n" :: [Char]
lines :: [Char] -> [[Char]]
words :: [Char] -> [[Char]]
unlines :: [[Char]] -> [Char]
unwords :: [[Char]] -> [Char]
(++) :: [Char] -> [Char] -> [Char]

                    words " " == lines ""
                     words "" == lines ""
                   words "\n" == lines ""
                     cs ++ "" == cs
                     "" ++ cs == cs
          words (unwords css) == words (unlines css)
           unlines (lines "") == ""
         unlines (lines "\n") == "\n"
          unwords (lines " ") == " "
           unwords (lines "") == ""
         unwords (lines "\n") == ""
                  " " ++ "\n" == unlines (lines " ")
unlines (lines (unlines css)) == unlines css
   lines (unlines (lines cs)) == lines cs
   lines (unlines (words cs)) == words cs
   words (unlines (lines cs)) == words cs
   words (unlines (words cs)) == words cs
            words (cs ++ " ") == words cs
           words (cs ++ "\n") == words cs
            words (" " ++ cs) == words cs
           words ("\n" ++ cs) == words cs

        "" <= cs
      "\n" <= " "
  lines "" <= css
lines "\n" <= lines " "

