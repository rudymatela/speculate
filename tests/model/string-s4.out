max expr size  =    4
  |- on ineqs  =    3
  |- on conds  =    3
max  #-tests   =  500
min  #-tests   =   25  (to consider p ==> q true)
max  #-vars    =    2  (for inequational and conditional laws)

_ :: [Char]
_ :: [[Char]]
"" :: [Char]
" " :: [Char]
"\n" :: [Char]
lines :: [Char] -> [[Char]]
words :: [Char] -> [[Char]]
unlines :: [[Char]] -> [Char]
unwords :: [[Char]] -> [Char]
(++) :: [Char] -> [Char] -> [Char]

                     cs ++ "" == cs
                     "" ++ cs == cs
unlines (lines (unlines css)) == unlines css
   lines (unlines (lines cs)) == lines cs
   words (unlines (words cs)) == words cs
   lines (unlines (words cs)) == words cs
            words (cs ++ " ") == words cs
           words (cs ++ "\n") == words cs
            words (" " ++ cs) == words cs
           words ("\n" ++ cs) == words cs
   words (unlines (lines cs)) == words cs
          words (unwords css) == words (unlines css)

                 "" <= cs
                 cs <= cs ++ ds
                 cs <= unlines (lines cs)
               "\n" <= cs ++ "\n"
               "\n" <= cs ++ " "
         "\n" ++ cs <= " "
 unwords (lines cs) <= cs ++ ds
         "\n" ++ cs <= ds ++ " "
 unlines (lines cs) <= cs ++ cs
         "\n" ++ cs <= cs ++ "\n"
         cs ++ "\n" <= cs ++ " "
 unlines (lines cs) <= cs ++ " "
 unlines (lines cs) <= cs ++ "\n"
           lines "" <= css
lines (unlines css) <= css

